// three/src/materials/nodes/NodeMaterials.js
// @TODO: We can simplify "export { default as SomeNode, other, exports } from '...'" to just "export * from '...'" if we will use only named exports

export { default as NodeMaterialObserver } from 'three/src/materials/nodes/manager/NodeMaterialObserver.js';

export { default as NodeMaterial } from 'three/src/materials/nodes/NodeMaterial.js';
export { default as LineBasicNodeMaterial } from 'three/src/materials/nodes/LineBasicNodeMaterial.js';
export { default as LineDashedNodeMaterial } from 'three/src/materials/nodes/LineDashedNodeMaterial.js';
export { default as Line2NodeMaterial } from 'three/src/materials/nodes/Line2NodeMaterial.js';
export { default as MeshNormalNodeMaterial } from 'three/src/materials/nodes/MeshNormalNodeMaterial.js';
export { default as MeshBasicNodeMaterial } from 'three/src/materials/nodes/MeshBasicNodeMaterial.js';
export { default as MeshLambertNodeMaterial } from 'three/src/materials/nodes/MeshLambertNodeMaterial.js';
export { default as MeshPhongNodeMaterial } from 'three/src/materials/nodes/MeshPhongNodeMaterial.js';
export { default as MeshStandardNodeMaterial } from 'three/src/materials/nodes/MeshStandardNodeMaterial.js';
export { default as MeshPhysicalNodeMaterial } from 'three/src/materials/nodes/MeshPhysicalNodeMaterial.js';
export { default as MeshSSSNodeMaterial } from 'three/src/materials/nodes/MeshSSSNodeMaterial.js';
export { default as MeshToonNodeMaterial } from 'three/src/materials/nodes/MeshToonNodeMaterial.js';
export { default as MeshMatcapNodeMaterial } from 'three/src/materials/nodes/MeshMatcapNodeMaterial.js';
export { default as PointsNodeMaterial } from 'three/src/materials/nodes/PointsNodeMaterial.js';
export { default as SpriteNodeMaterial } from 'three/src/materials/nodes/SpriteNodeMaterial.js';
export { default as ShadowNodeMaterial } from 'three/src/materials/nodes/ShadowNodeMaterial.js';
export { default as VolumeNodeMaterial } from 'three/src/materials/nodes/VolumeNodeMaterial.js';
