// three/src/materials/Materials.js
import { ShadowMaterial } from 'three/src/materials/ShadowMaterial.js';
import { SpriteMaterial } from 'three/src/materials/SpriteMaterial.js';
import { RawShaderMaterial } from 'three/src/materials/RawShaderMaterial.js';
import { ShaderMaterial } from 'three/src/materials/ShaderMaterial.js';
import { PointsMaterial } from 'three/src/materials/PointsMaterial.js';
import { MeshPhysicalMaterial } from 'three/src/materials/MeshPhysicalMaterial.js';
import { MeshStandardMaterial } from 'three/src/materials/MeshStandardMaterial.js';
import { MeshPhongMaterial } from 'three/src/materials/MeshPhongMaterial.js';
import { MeshToonMaterial } from 'three/src/materials/MeshToonMaterial.js';
import { MeshNormalMaterial } from 'three/src/materials/MeshNormalMaterial.js';
import { MeshLambertMaterial } from 'three/src/materials/MeshLambertMaterial.js';
import { MeshDepthMaterial } from 'three/src/materials/MeshDepthMaterial.js';
import { MeshDistanceMaterial } from 'three/src/materials/MeshDistanceMaterial.js';
import { MeshBasicMaterial } from 'three/src/materials/MeshBasicMaterial.js';
import { MeshMatcapMaterial } from 'three/src/materials/MeshMatcapMaterial.js';
import { LineDashedMaterial } from 'three/src/materials/LineDashedMaterial.js';
import { LineBasicMaterial } from 'three/src/materials/LineBasicMaterial.js';
import { Material } from 'three/src/materials/Material.js';

export {
	ShadowMaterial,
	SpriteMaterial,
	RawShaderMaterial,
	ShaderMaterial,
	PointsMaterial,
	MeshPhysicalMaterial,
	MeshStandardMaterial,
	MeshPhongMaterial,
	MeshToonMaterial,
	MeshNormalMaterial,
	MeshLambertMaterial,
	MeshDepthMaterial,
	MeshDistanceMaterial,
	MeshBasicMaterial,
	MeshMatcapMaterial,
	LineDashedMaterial,
	LineBasicMaterial,
	Material
};
